lib/fog/storage/google_json/models/file.rb

```diff
@@ -52,6 +70,15 @@ def destroy
           false
         end
 
+        def public=(new_public)
+          if new_public
+            @predefined_acl = "publicRead"
+          else
```

Comment:
Please add a comment to this function. `public = false` could overwrite permissions that people don't expect to change.